src/card/color.rs

```diff
@@ -25,41 +25,142 @@ pub enum Color {
+    pub const ESPER: Self = Colors::colorless().white().blue().black();
+    /// Grixi (Blue, Black, Red)
+    pub const GRIXIS: Self = Colors::colorless().blue().black().red();
+    /// Jund (Black, Red, Green)
+    pub const JUND: Self = Colors::colorless().black().red().green();
+    /// Naya (White, Red, Green)
+    pub const NAYA: Self = Colors::colorless().white().red().green();
+    /// Abzan (White, Black, Green)
+    pub const ABZAN: Self = Colors::colorless().white().black().green();
+    /// Jeskai (White, Blue, Red)
+    pub const JESKAI: Self = Colors::colorless().white().blue().red();
+    /// Sultai (Blue, Black, Green)
+    pub const SULTAI: Self = Colors::colorless().blue().black().green();
+    /// Mardu (White, Black, Red)
+    pub const MARDU: Self = Colors::colorless().white().black().red();
+    /// Temur (Blue, Red, Green)
+    pub const TEMUR: Self = Colors::colorless().blue().red().green();
+    /// Chaos (White, Blue, Black, Red)
+    pub const CHAOS: Self = Colors::colorless().white().blue().black().red();
+    /// Artifice (Blue, Black, Red, Green)
+    pub const ARTIFICE: Self = Colors::colorless().blue().black().red().green();
+    /// Aggression (Black, Red, Green, White)
+    pub const AGGRESSION: Self = Colors::colorless().black().red().green().white();
+    /// Altruism (Red, Green, White, Blue)
+    pub const ALTRUISM: Self = Colors::colorless().red().green().white().blue();
+    /// Growth (Green, White, Blue, Black)
+    pub const GROWTH: Self = Colors::colorless().green().white().blue().black();
+    /// WUBRG (White, Blue, Black, Red, Green)
+    pub const WUBRG: Self = Colors::colorless().white().blue().black().red().green();
+
     /// Creates an instance representing a multicolored card without specifying
     /// its colors.
-    pub fn multicolored() -> Self {
+    pub const fn multicolored() -> Self {
         Colors(1 << 7)
     }
 
     /// Creates an instance representing a colorless card.
-    pub fn colorless() -> Self {
+    pub const fn colorless() -> Self {
         Colors(Color::Colorless as u8)
     }
 
     /// Checks to see if a card is a certain color.
     ///
-    /// Note: Multicolored cards are may not be any particular color.
-    pub fn is(self, color: Color) -> bool {
+    /// Note: Multicolored cards may not be any particular color.
+    pub const fn is(self, color: Color) -> bool {
         self.0 & color as u8 != 0
     }
 
     /// Checks if a card is multicolored. This only works for instances
     /// created by [`Colors::multicolored`].
     ///
     /// [`Colors::multicolored`]: #method.multicolored
-    pub fn is_multicolored(self) -> bool {
+    pub const fn is_multicolored(self) -> bool {
         self.0 & (1 << 7) != 0
     }
 
     /// Checks if a card is colorless.
-    pub fn is_colorless(self) -> bool {
+    pub const fn is_colorless(self) -> bool {
         self.0 == 0
     }
+
+    /// Adds white
+    pub const fn white(mut self) -> Self {
```

Comment:
TBH I'm not a fan of this API. It reads really awkwardly to me. I'd prefer something like, `fn with(self, color: Color)` (or `add`).

Reply:
Well the add one is there and this way you can just write the colors you want
```
Colors::colorless().white().blue();
```
vs
```
Colors::colorless().with(Color::White).with(Color::Blue)
```
it just felt the "aliases" where worth it.

Reply:
Maybe you could use `Default`?

Reply:
Yeah I like that

Reply:
Ah, `default` isn't const

Reply:
Did you see what I did in my branch?

Reply:
I did. It's kinda the same thing, just done on the search side, right?

Reply:
It is, and the aliases are defined in enums. Another addition is the `const fn from_slice` which loops over the colors in a slice, so if you were to take the associated constants approach you'd write: `pub const AZORIUS: Self = Colors::from_slice(&[Blue, White]);`

Overall I think it makes sense to put the aliases in the same module as `Colors` instead of a submodule of `search`. The approaches of using associated constants vs enums each have their pros and cons.

Reply:
Another thing I was thinking about was whether it makes sense for Colors to have the multicolor variant. With the new search module, we could make the unit struct `pub struct Multicolored;`, and have it implement `ColorValue`. Searching with m plus other colors is [unsupported anyway](https://scryfall.com/search?q=color%3E%3Dbm).

Reply:
:thinking: writting this extra code I was a bit confused as to why I had multicolored here, now it makes sense but I agree that it's a bad idea.

Reply:
Enum vs associated constants, why not both? Too much bloat maybe? Enums are iterable which is nice but this could be provided by a function. What other arguments are there in favor of enums?

Reply:
I'm a bit afraid this might bloat the docs, you could use the new doc alias macro instead `#[doc(alias = "wb")]` (it's case insensitive) this way if someone searches for `"wb"`, azorius will show up.

Reply:
That's a fair point. It's probably better to use the nicknames only. Or I could keep the initial constants (e.g. 'WR`, `WBG`) around and use `#[doc(hidden)]` for them.